### PR TITLE
Ilof pietro

### PR DESCRIPTION
**Pull request by pietro-tanure**
ilof.py: I have created the Incremental Local Outlier Factor class that can be used further to other ILOF variations like DILOF, MILOF, GILOF

__init__.py    I added it to the list of anomaly algorithms for river

utils/math.py:     It took me a long time but I realized that the mikowski distance was defined wrongly, which was giving bad results to my algorithm, so I propose a correction

### CODE REVIEW

File: river/utils/math.py [hunk 1]
```diff
@@ -160,7 +160,7 @@ def minkowski_distance(a: dict, b: dict, p: int):
         Manhattan distance. When `p=2`, this is equivalent to using the Euclidean distance.
 
     """
-    return sum((abs(a.get(k, 0.0) - b.get(k, 0.0))) ** p for k in {*a.keys(), *b.keys()})
+    return sum((abs(a.get(k, 0.0) - b.get(k, 0.0))) ** p for k in {*a.keys(), *b.keys()})**(1/p)
```
Comment (MaxHalford) [inline river/utils/math.py:163]: Wow, good catch!

**Comment by pietro-tanure** [comment]
I created a learn_many function that takes pd.Series as you advised

**Comment by pietro-tanure** [comment]
@MaxHalford I just ran the pre commits and set everything in order now there's not errors in my files, the precommits did automatically change a few things on some other codes, but the ones related to my project are just 3:
anomaly/ilof.py
anomaly/base.py
utils/math.py

**Comment by MaxHalford** [comment]
Thanks @pietro-tanure!

**Comment by hoanganhngo610** [comment]
Hi @pietro-tanure! Since this pull request is quite inactive for quite some time, I would be eager to continue the review so that the model can be added to River ASAP!

**Comment by pietro-tanure** [comment]
> Hi @pietro-tanure! Since this pull request is quite inactive for quite some time, I would be eager to continue the review so that the model can be added to River ASAP!

Hello @hoanganhngo610, thank you, do I have to give some type of permission on the repository or pull request?

**Comment by hoanganhngo610** [comment]
@pietro-tanure I don't think there would be any permission needed. I would have a first pass through the repository, and then will make a PR to your branch if needed. In the meantime, it would be extremely appreciated if you can resolve conflicting files with the newest version of `River` available.

**Pull request closed without merging**